### DISCUSSION TITLE
Invariant Violation: requireNativeComponent: "RNSScreen" was not found in the UIManager

### DISCUSSION THREAD

**Discussion by lfeldman65** [react-navigation/react-navigation#10575] [Q&A] [answered]
I am getting RNSScreen error even though I have installed all related packages and followed react-navigation guide (https://reactnavigation.org/docs/getting-started#installation) but nothing worked for me.  Your help is appreciated!  Thanks!

Note:  I don't get this error when running on a simulator or running on an iPhone with the QR code.  The error (and crash) only occurs in production mode.

![image](https://user-images.githubusercontent.com/12520398/167513464-82d57743-bef3-42b2-98f2-fff23cf76c47.png)

![image](https://user-images.githubusercontent.com/12520398/167513483-a4ac28df-3958-481c-ba01-654128c924e6.png)

![image](https://user-images.githubusercontent.com/12520398/167513488-b5a3eea6-05a0-4359-aa4e-b84be98cb481.png)

**Comment by raajnadar** [comment]
Delete .lock file and re install it. After that clear metro cache and start the application.

**Reply by lfeldman65** [reply]
Thank you for your reply!

Delete the yarn.lock or package.lock or both?

Sorry for the noob question, but how do I clear the metro cache?

Thanks again!
Larry

P.S.  I just noticed that I left some important information out of my description.  Sorry about that.

I don't get this error when running on a simulator or running on an iPhone with the QR code.  The error (and crash) only occurs in production mode.

More supporting info: https://stackoverflow.com/questions/72104735/expo-with-drawer-navigator-crashes-on-ios-testflight-app-requirenativecomponen

**Reply by raajnadar** [reply]
Run cmd `expo upgrade 44.0.0` you are using non compatible packages, follow the questionnaires given by the cli to solve the issue.

**Reply by raajnadar** [reply]
Run `expo start -c` to clear the metro cache with expo, non expo `npx react-native start --reset-cache`

**Comment by lfeldman65** [comment, marked as answer]
I updated Expo to 45.0.0. That fixed the error and solved a problem with the app crashing after launch in production mode but working fine on device with QR code or simulator.

**Reply by raajnadar** [reply]
As I told the packages were not compatible with SDK 44, upgrade solved the issue.

**Comment by lfeldman65** [comment]
You’re correct that it was a package version issue, but you advised me to update expo to 44.0.0. I had to update expo to 45.0.0. I appreciate your help!

Larry

**Reply by raajnadar** [reply]
Yes that is great you upgraded to latest some people are still not read to upgrade to latest version so I gave you your current version so that there is no breaking changes and issue gets solved.

**Comment by lfeldman65** [comment]
If people aren’t ready to upgrade to latest, how will they make a TestFlight build that doesn’t crash? I think there was a bug in Expo 44.0.0. 

Larry

**Reply by raajnadar** [reply]
I was using SDK 44 since 5 months never got a crash issue, you might have messed up some configuration, which got solved with `expo upgrade` cmd

**Comment by lfeldman65** [comment]
Could be. Thanks!

Larry
